qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -0,0 +1,322 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Quantum Kernel Algorithm"""
+
+from typing import Optional, Union
+
+import logging
+
+import numpy as np
+
+from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+from qiskit.circuit import ParameterVector
+from qiskit.providers import Backend, BaseBackend
+
+from qiskit.utils import QuantumInstance
+from qiskit.algorithms import AlgorithmError
+
+logger = logging.getLogger(__name__)
+
+
+class QuantumKernel:
+    r"""Quantum Kernel.
+
+    The general task of machine learning is to find and study patterns in data. For many
+    algorithms, the datapoints are better understood in a higher dimensional feature space,
+    through the use of a kernel function:
+
+    .. math::
+        K(x, y) = \langle f(x), f(y)\rangle.
+
+    Here K is the kernel function, x, y are n dimensional inputs. f is a map from n-dimension
+    to m-dimension space. :math:`\langle x, y \rangle` denotes the dot product.
+    Usually m is much larger than n.
+
+    The quantum kernel algorithm calculates a kernel matrix, given datapoints x and y and feature
+    map f, all of n dimension. This kernel matrix can then be used in classical machine learning
+    algorithms such as support vector classification, spectral clustering or ridge regression.
+    """
+
+    def __init__(self,
+                 feature_map: QuantumCircuit,
```

Comment:
Often we will default something like this (the feature_map in this case) to None and internally create a default. We state what the default is of course. If there is no useful./reasonable default then requiring one seems fine. But it also relates to the comments I made about SVC above.

Reply:
@stefan-woerner What do you think? I'm not sure what a good default feature map could be.

Reply:
We can use the `ZZFeatureMap` as default, that's the one from the original paper.

Reply:
Default `ZZFeatureMap` added with default 2 parameters.